config_scheduler.go

```diff
@@ -46,47 +54,87 @@ func (c *SchedulerConfig) Set() error {
 
 	_, err := coll.UpdateOne(ctx, byId(c.SectionId()), bson.M{
 		"$set": bson.M{
-			"task_finder":        c.TaskFinder,
-			"host_allocator":     c.HostAllocator,
-			"free_host_fraction": c.FreeHostFraction,
+			"task_finder":                       c.TaskFinder,
+			"host_allocator":                    c.HostAllocator,
+			"free_host_fraction":                c.FreeHostFraction,
+			"planner_version":                   c.PlannerVersion,
+			"target_time_seconds":               c.TargetTimeSeconds,
+			"acceptable_host_idle_time_seconds": c.AcceptableHostIdleTimeSeconds,
+			"group_versions":                    c.GroupVersions,
+			"patch_zipper_factor":               c.PatchZipperFactor,
+			"interleave":                        c.Interleave,
+			"mainline_first":                    c.MainlineFirst,
+			"patch_first":                       c.PatchFirst,
 		},
 	}, options.Update().SetUpsert(true))
 
 	return errors.Wrapf(err, "error updating section %s", c.SectionId())
 }
 
 func (c *SchedulerConfig) ValidateAndDefault() error {
-	finders := []string{"legacy", "alternate", "parallel", "pipeline"}
-
 	if c.TaskFinder == "" {
 		// default to legacy
-		c.TaskFinder = finders[0]
+		c.TaskFinder = FinderVersionLegacy
 		return nil
 	}
 
 	if c.CacheDurationSeconds == 0 || c.CacheDurationSeconds > 600 {
 		c.CacheDurationSeconds = 20
 	}
 
-	if !util.StringSliceContains(finders, c.TaskFinder) {
+	if !util.StringSliceContains(ValidFinderVersions, c.TaskFinder) {
 		return errors.Errorf("supported finders are %s; %s is not supported",
-			finders, c.TaskFinder)
+			ValidFinderVersions, c.TaskFinder)
 	}
 
-	allocators := []string{"duration", "deficit", "utilization"}
 	if c.HostAllocator == "" {
-		c.HostAllocator = allocators[0]
+		c.HostAllocator = HostAllocatorDuration
```

Comment:
Should the planner version have a default? Since it's a new field we won't have it set the first time we start the app after deploying

Reply:
Done.